app/android/src/uk/co/lutraconsulting/CameraActivity.java

```diff
@@ -73,85 +100,240 @@ protected void onCreate(Bundle savedInstanceState) {
+                    SENSOR_DELAY);
+        }
+        if (mSensorMagnetometer != null) {
+            mSensorManager.registerListener(this, mSensorMagnetometer,
+                    SENSOR_DELAY);
+        }
+    }
+
+    @Override
+    public void onSensorChanged(SensorEvent sensorEvent) {
+        // The sensor type (as defined in the Sensor class).
+        int sensorType = sensorEvent.sensor.getType();
+
+        // The sensorEvent object is reused across calls to onSensorChanged().
+        // clone() gets a copy so the data doesn't change out from under us
+        switch (sensorType) {
+            case Sensor.TYPE_ACCELEROMETER:
+                mAccelerometerData = sensorEvent.values.clone();
+                break;
+            case Sensor.TYPE_MAGNETIC_FIELD:
+                mMagnetometerData = sensorEvent.values.clone();
+                break;
+            default:
+                return;
+        }
+        // Compute the rotation matrix: merges and translates the data
+        // from the accelerometer and magnetometer, in the device coordinate
+        // system, into a matrix in the world's coordinate system.
+        //
+        // The second argument is an inclination matrix, which isn't
+        // used in this example.
+        float[] rotationMatrix = new float[9];
+        boolean rotationOK = SensorManager.getRotationMatrix(rotationMatrix,
+                null, mAccelerometerData, mMagnetometerData);
+
+        // Remap the matrix based on current device/activity rotation.
+        float[] rotationMatrixAdjusted = new float[9];
+        rotationMatrixAdjusted = rotationMatrix.clone();
+
+        // Get the orientation of the device (azimuth, pitch, roll) based
+        // on the rotation matrix. Output units are radians.
+        float orientationValues[] = new float[3];
+        if (rotationOK) {
+            SensorManager.getOrientation(rotationMatrixAdjusted,
+                    orientationValues);
+        }
+
+        // Pull out the individual values from the array.
+        float azimuth = orientationValues[0];
+        float pitch = orientationValues[1];
+        float roll = orientationValues[2];
+
+        // angle in degree [0 - 360] degree
+        double degrees = toDegrees(orientationValues[0]); // angle of rotation about the -z axis
+        double pitchDegrees = toDegrees(orientationValues[1]); // angle of rotation about the x axis
+        double rollDegrees = toDegrees(orientationValues[2]); // angle of rotation about the y axis
+
+        // Store sensor values
+        double normDegrees = adjustDegreesToScreenOrientation(degrees);
+        double cameraAngle = adjustCameraAngleOrientation(pitchDegrees, rollDegrees);
+
+        azimuthData.put(System.currentTimeMillis(), normDegrees);
+        pitchData.put(System.currentTimeMillis(), pitchDegrees);
+        rollData.put(System.currentTimeMillis(), rollDegrees);
+        cameraAngleData.put(System.currentTimeMillis(), cameraAngle);
```

Comment:
I am wondering if ~0.2 ms interval is not too high, witch unnecessarily too many numbers stored. Maybe 5ms could work too? That should be still a plenty of data to use...